node_modules/oae-core/signin/js/signin.js

```diff
@@ -0,0 +1,191 @@
+        var doLocalSignIn = function() {
+            // Hide the login button and show a logging in message
+            var signingInMessage = oae.api.i18n.translate('__MSG__SIGNING_IN__');
+            $('#signin-local-form button', $rootel).html(signingInMessage);
+            $('#signin-local-form *', $rootel).prop('disable', 'true');
+
+            // Get the entered username and password
+            var username = $.trim($('#signin-local-username', $rootel).val());
+            var password = $.trim($('#signin-local-password', $rootel).val());
+
+            var ldapEnabled = authStrategyInfo[oae.api.authentication.STRATEGY_LDAP];
+            var localEnabled = authStrategyInfo[oae.api.authentication.STRATEGY_LOCAL];
+
+            // Both LDAP and local are enabled. We try LDAP first, and try local next if LDAP has failed
+            if (ldapEnabled && localEnabled) {
+                oae.api.authentication.LDAPLogin(username, password, function(err) {
+                    if (err) {
+                        oae.api.authentication.localLogin(username, password, finishLogin);
+                    } else {
+                        finishLogin();
+                    }
+                });
+            // Only LDAP is enabled
+            } else if (ldapEnabled) {
+                oae.api.authentication.LDAPLogin(username, password, finishLogin);
+            // Only local authentication is enabled
+            } else {
+                oae.api.authentication.localLogin(username, password, finishLogin);
+            }
+        };
+
+        /**
+         * Bind the validation logic to the local sign in form
+         */
+        var validateLocalSignInForm = function() {
+            var validateOpts = {
+                'messages': {
+                    'signin-local-username': {
+                        'required': oae.api.i18n.translate('__MSG__PLEASE_ENTER_YOUR_USERNAME__'),
+                        'signin-local-failed-attempt': oae.api.i18n.translate('__MSG__INVALID_USERNAME_OR_PASSWORD__')
+                    },
+                    'signin-local-password': oae.api.i18n.translate('__MSG__PLEASE_ENTER_YOUR_PASSWORD__')
+                },
+                'methods': {
+                    'signin-local-failed-attempt': {
+                        'method': function(value, element) {
+                            // This class will be added after a failed login attempt and is used
+                            // to tell jquery.validate to mark the field as invalid
+                            return false;
+                        }
+                    }
+                },
+                'submitHandler': doLocalSignIn
+            };
+
+            oae.api.util.validation().validate($('#signin-local-form', $rootel), validateOpts);
+        }
+
+        /**
+         * Render the sign in options (external auth options, username & password form, etc...)
+         * form
+         */
+        var renderSignInOptions = function() {
```

Comment:
When local authentication is enabled, focus should be set to the username field when the signin modal is launched